sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -441,10 +447,10 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
             DoFn.InputProvider.class.getSimpleName(),
             formatType(paramT),
             formatType(expectedInputProviderT));
-        extraParameters.add(DoFnSignature.Parameter.INPUT_PROVIDER);
+        extraParameters.add(DoFnSignature.Parameter.inputProvider());
```

Comment:
Done. In fact it is not so much a requirement as a provision - `Parameter` is closed-world. The ones that exist are the only ones that can exist.